compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -172,7 +172,38 @@ object Types {
         case _ =>
           false
       }
-      cls == defn.AnyClass || loop(this)
+      loop(this)
+    }
+
+    /** Returns the top type of the lattice
+     *   - XYX.Any if XYZ extends scala.Phantom and this type is upper bounded XYZ.Any
+     *   - scala.Any otherwise
+     */
+    final def topType(implicit ctx: Context): TypeRef = {
+      val lattice = phantomLatticeType
```

Comment:
In fact `phantomLatticeType` goes to the to but reaches the `ClassInfo` of the `Phantom.Any` where I require a `TypeRef`. Therefore still need to select the `Any` on the prefix of that first top found.
Originally, the code was like that, but I extracted `phantomLatticeType` because I noticed that it was possible to have a more efficient implementation of `bottomType` and `isPhantom`.